internal/version/client.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package version
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/transport"
+)
+
+// VersionedClient cointains and managers a transport client, that internally checks if the software version
+// is still up to date and refuses outgoing connections in case it is outdated
+type VersionedClient struct {
```

Comment:
i'd like to take this out of this PR if that's okay

Reply:
Will do that 👍